services/121-service/src/payments/fsp-integration/nedbank/nedbank-api.service.spec.ts

```diff
@@ -0,0 +1,273 @@
+        },
+      });
+
+      // Check the headers
+      expect(requestCallParamObject.headers).toEqual([
+        { name: 'x-ibm-client-id', value: process.env.NEDBANK_CLIENT_ID },
+        {
+          name: 'x-ibm-client-secret',
+          value: process.env.NEDBANK_CLIENT_SECRET,
+        },
+        { name: 'x-idempotency-key', value: expect.any(String) },
+        { name: 'x-jws-signature', value: expect.any(String) },
+        { name: 'x-fapi-financial-id', value: 'OB/2017/001' },
+        { name: 'x-fapi-customer-ip-address', value: process.env.PUBLIC_IP },
+        { name: 'x-fapi-interaction-id', value: expect.any(String) },
+        { name: 'Content-Type', value: 'application/json' },
+      ]);
+
+      expect(requestCallParamObject.method).toBe('POST');
+    });
+
+    it('should throw an error if create order fails', async () => {
+      const errorResponse: AxiosResponse<ErrorReponseNedbankDto> = {
+        data: {
+          Message: 'BUSINESS ERROR',
+          Code: 'NB.APIM.Field.Invalid',
+          Id: '1d3e3076-9e1c-4933-aa7f-69290941ec70',
+          Errors: [
+            {
+              ErrorCode: 'NB.APIM.Field.Invalid',
+              Message:
+                'Request Validation Error - TPP account configuration mismatch',
+              Path: '',
+              Url: '',
+            },
+          ],
+        },
+        status: 201, // Nedbank returns 201 even on errors
+        statusText: '',
+        headers: {},
+        config: {},
+      };
+
+      jest.spyOn(httpService, 'request').mockResolvedValue(errorResponse);
+
+      await expect(
+        service.createOrder({
+          transferAmount: amount,
+          phoneNumber: registrationNedbank.phoneNumber,
+          orderCreateReference,
+        }),
+      ).rejects.toThrow(NedbankError);
+
+      // TODO: Not sure if this is the best/prettiest syntax to test the content of the error but it works
+      let errorOnCreateOrder: NedbankError | undefined;
+      try {
+        await service.createOrder({
+          transferAmount: amount,
+          phoneNumber: registrationNedbank.phoneNumber,
+          orderCreateReference,
+        });
+      } catch (error) {
+        errorOnCreateOrder = error;
+      }
+      expect(errorOnCreateOrder).toMatchSnapshot();
+    });
+
+    it('should throw an error if httpsAgent is not defined', async () => {
+      service.httpsAgent = undefined;
+
+      // TODO: Not sure if this is the best/prettiest syntax to test the content of the error but it works
+      let errorOnCreateOrder: NedbankError | undefined;
+      try {
+        await service.createOrder({
+          transferAmount: amount,
+          phoneNumber: registrationNedbank.phoneNumber,
+          orderCreateReference,
+        });
+      } catch (error) {
+        errorOnCreateOrder = error;
+      }
+      expect(errorOnCreateOrder).toMatchSnapshot();
+    });
+  });
+
+  describe('getOrder', () => {
```

Comment:
Did you consider unit testing edge cases for the input parameter? Like an empty string, or very long text?